docs/DeveloperGuide.adoc

```diff
@@ -96,171 +109,383 @@ The `UI` component,
 
-* can save `UserPref` objects in json format and read it back.
-* can save the Address Book data in json format and read it back.
+. Saves `UserPref` objects in json format and read it back for the next session.
+. Saves `FoodRecord` data in json format and read it back for the next session.
+. Generates an insights report based on user food consumption.
+. Generates a user-friendly version of the current `FoodRecord`.
 
 [[Design-Commons]]
 === Common classes
 
-Classes used by multiple components are in the `seedu.addressbook.commons` package.
+Classes used by multiple components are in the `life.calgo.commons` package.
+
+<<<
 
 == Implementation
 
 This section describes some noteworthy details on how certain features are implemented.
 
-// tag::undoredo[]
-=== [Proposed] Undo/Redo feature
-==== Proposed Implementation
+// tag::lexicographicallistcommand[]
+=== Lexicographically-Ordered `list`
+==== Implementation
+With a large number of `Food` objects in the App, it will be useful to see all entries sorted in lexicographic order. +
 
-The undo/redo mechanism is facilitated by `VersionedAddressBook`.
-It extends `AddressBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`.
-Additionally, it implements the following operations:
+The `list` command is able to show all current `Food` entries in the `Model`.
+Therefore, it is one of the most commonly used commands in `Calgo`, especially after a `find` command.
+It may seem that the `list` command is responsible for keeping the `Food` entries sorted, however, that is not the case.
 
-* `VersionedAddressBook#commit()` -- Saves the current address book state in its history.
-* `VersionedAddressBook#undo()` -- Restores the previous address book state from its history.
-* `VersionedAddressBook#redo()` -- Restores a previously undone address book state from its history.
+Sorting `Food` objects by lexicographic order is done by the `UniqueFoodList`.
+`Food` objects are able to be sorted because the `Food` class implements the `Comparable<Food>` interface. +
 
-These operations are exposed in the `Model` interface as `Model#commitAddressBook()`, `Model#undoAddressBook()` and `Model#redoAddressBook()` respectively.
+How the `UniqueFoodList` works:
 
-Given below is an example usage scenario and how the undo/redo mechanism behaves at each step.
+* Existing `Food` objects in the list are arranged in lexicographic order by `Name`.
+* `UniqueFoodList` sorts the `Food` whenever they are added or edited in the `Model`.
 
-Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
+With this, a top-level idea of the execution of the lexicographical `list` command is given in the sequence diagram below:
 
-image::UndoRedoState0.png[]
+image::ListCommandSequenceDiagram.png[]
```

Comment:
As the diagram is large, maybe you can consider putting updateFilteredFoodRecord() as a method which has a return arrow?